pkg/api/v1alpha1/cluster_webhook_test.go

```diff
@@ -885,7 +885,7 @@ func TestClusterValidateUpdateIdentityProviderRefsImmutableKind(t *testing.T) {
 	}
 
 	g := NewWithT(t)
-	g.Expect(c.ValidateUpdate(cOld)).NotTo(Succeed())
+	g.Expect(c.ValidateUpdate(cOld)).To(Succeed())
```

Comment:
you can add a test to validate awsiamconfig is still immutable

Reply:
Added those here https://github.com/mitalipaygude/eks-anywhere-1/blob/update_oidc/pkg/validations/upgradevalidations/preflightvalidations_test.go#L182

Reply:
they are different tho. webhook is for controller, preflight validation is for CLI.

Reply:
Added the change